src/utils/execWithOutput.js

```diff
@@ -3,6 +3,28 @@
 const { StringDecoder } = require('node:string_decoder')
 
 const { exec } = require('@actions/exec')
+const { CONFIDENTIAL_KEYWORDS_FOR_REDACTION } = require('../const')
+
+/**
+ * 
+ * @param {string[]} args 
+ * @returns Redacted Array or Blank Array if null/undefined
+ */
+function removeConfidentialArguments(args) {
+  let skipItem = false
+
+  return (args ?? []).filter(arg => {
+      if (skipItem) {
+          skipItem = false
+          
+          return false;
+      }
+
+      skipItem = CONFIDENTIAL_KEYWORDS_FOR_REDACTION.includes(arg?.toString().toLocaleUpperCase())
+
+      return !skipItem
+  })
+}
```

Comment:
it's not very easy to follow what this code is doing, plus there are not tests for this function outside of the edge cases. let's please write this in a more idiomatic way so that understanding what it does it easier for a reader